frontends/p4/strengthReduction.h

```diff
@@ -26,13 +26,37 @@ limitations under the License.
 
 namespace P4 {
 
+/** Implements a pass that replaces expensive arithmetic and boolean
+  * operations with cheaper ones -- i.e., strength reduction
+  * 
+  * Specifically, it provides:
+  * 
+  * 1. A collection of helper methods that determine whether a given
+  * expression is `0`, `1`, `true`, or `false`, or a power of `2`
+  * 
+  * 2. A visitor that transforms arithmetic and boolean exprssions
+  * 
+  * @pre: None
+  * 
+  * @post: 
+  *   - all arithmetic and boolean expressions are simplified
+  *   - check for div / mod by `0`
+  */ 
+// TODO:
+//   - overlap with constant folding?
+//   - store constant values in IR?
 class StrengthReduction final : public Transform {
+    /// Helper that returns `true` if `expr` is the constant `1`.
     bool isOne(const IR::Expression* expr) const;
+    /// Helper that returns `true` if `expr` is the constant `0.
     bool isZero(const IR::Expression* expr) const;
-    bool isFalse(const IR::Expression* expr) const;
+    /// Helper that returns `true` if `expr` is the constant `true`.
     bool isTrue(const IR::Expression* expr) const;
-    // if expr is a constant value that is a power of 2 return the log2 of it,
-    // else return -1
+    /// Helper that returns `true` if `expr` is the constant `false`.
+    bool isFalse(const IR::Expression* expr) const;
+    /** Helper that returns the logarithm (base 2) of `expr` if it is
+     * positive and a power of `2`, or `-1` otherwise. 
```

Comment:
It's fine both ways, but just as an FYI, another option you have to write this in Doxygen is:

```
/// Helper that returns the logarithm (base 2) of `expr` if it is
/// positive and a power of `2`, or `-1` otherwise. 
```

In other words, consecutive lines that start with `///` will be joined.

In the past, I've often used the rule of thumb that for one or two lines, `///` is more readable, and for three or more `/** ... */` is more readable. It's a matter of taste, though.